### PR TITLE
Implement a bunch of `From`s

### LINKED ISSUE #241: Implement From on Value for more types
`Value` implements `From` for numeric types. It should also implement `From` for any type that can be represented as a JSON value, such as `Map<String, Value>`, `bool`, `Vec<Value>` (and probably `&'a [Value]` as well), and in particular, for `String`, `&'a str`, and `Cow<'a, str>`. That way I can write code like

```rust
map.insert("key".into(), someStr.into())
```

instead of

```rust
map.insert("key".into(), serde_json::Value::String(someStr.into()))
```

### PR DESCRIPTION
**Pull request by killercup**
This is how I spend my lunch breaks! 😆 (Mostly copy-pasta, though.)

I'd love to `impl<T: Into<String>> From<T> for Value` and probably also `impl<T: Iterator<Item=I>, I: Into<Value> From<T> for Value`. Maybe with specialization?

Fixes #241

### CODE REVIEW

File: json/src/value.rs [hunk 1]
```diff
@@ -812,17 +813,181 @@ from_integer! {
 }
 
 impl From<f32> for Value {
+    /// Convert 32-bit floating point number to `Value`
```
Comment (oli-obk) [inline json/src/value.rs]: do documented impls show up in rustdoc at all?

Comment (killercup) [inline json/src/value.rs]: Yes, likes this:

bildschirmfoto 2017-02-13 um 15 03 16

(Also, the doc tests are nicer than unit tests here IMHO.)

**Comment by oli-obk** [comment]
> This is how I spend my lunch breaks!

/me hides shamefully for being lazy

File: json/src/value.rs [hunk 2]
```diff
@@ -812,17 +813,181 @@ from_integer! {
...
+    /// Convert string slice to `Value`
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # extern crate serde_json;
+    /// use serde_json::Value;
+    ///
+    /// # fn main() {
+    /// let s: String = "lorem".to_string();
```
Comment (oli-obk) [inline json/src/value.rs]: too much pasta

Comment (killercup) [inline json/src/value.rs]: Oops. I meant "ipsum", obviously :trollface:

Comment (killercup) [inline json/src/value.rs]: Fixed. Kept the "lorem", though.

**Comment by oli-obk** [comment]
Do you think it would be totally crazy to implement [`FromIterator<T>`](https://doc.rust-lang.org/std/iter/trait.FromIterator.html) for `Value`, yielding `Value::Array` for `T: Into<Value>`?

Then we could do `someiterator.collect()` and get a `Value::Array`. Seems cleaner to me than that `Clone` bound.

**Comment by killercup** [comment]
@oli-obk, "crazy", you say? "DONE!", I answer! I did not remove the Vec and slice impl, though, as they are still a bit more ergonomic and you don't need to `use std::iter::FromIterator;`.

Feel free to pick and choose what you'd like to have :)

**Pull request merged**